odriveS1/src/odriveS1/odriveCAN.py

```diff
@@ -0,0 +1,263 @@
+            if msg.arbitration_id == ((axisID << 5) | db.get_message_by_name('Heartbeat').frame_id):
+                current_state = db.decode_message('Heartbeat', msg.data)['Axis_State']
+                if (current_state != 0x0) & (current_state != 0x1):
+                    msg = db.get_message_by_name('Get_Encoder_Count')
+                    data1 = msg.encode({'Shadow_Count': 0, 'Count_in_CPR': 4000})
+                    msg = can.Message(arbitration_id=0x00A | axisID << 5, is_extended_id=False, data=data1)
+                    await self.try_send(msg)
+
+                    for msg1 in bus:
+                        if msg1.arbitration_id == ((axisID << 5) | 0x009):
+                            encoderCount = db.decode_message('Get_Encoder_Estimates', msg1.data)
+                            current_power = encoderCount['Vel_Estimate']/(self.max_rpm/60)
+                            return [True, current_power]
+                else:
+                    return [False, 0]
+
+    
+    async def is_moving(self) -> bool:
+        for msg in bus:
+            if msg.arbitration_id == ((axisID << 5) | db.get_message_by_name('Heartbeat').frame_id):
+                current_state = db.decode_message('Heartbeat', msg.data)['Axis_State']
+                if (current_state != 0x0) & (current_state != 0x1):
+                    return True
+                else:
+                    return False
+                
+    async def do_command(self) -> Dict[str, Any]:
+        pass
+
+    async def wait_until_correct_state(self, state):
+        for msg in bus:
+            current_state = db.decode_message('Heartbeat', msg.data)['Axis_State']
+            if current_state == state:
+                return
+   
+    async def wait_and_set_to_idle(self, goal):
+        while True:
+            position = await self.get_position()
+            if abs(position - goal) <= 0.01:
+                msg = db.get_message_by_name('Set_Axis_State')
+                data = msg.encode({'Axis_Requested_State': 0x01})
+                msg = can.Message(arbitration_id=msg.frame_id | axisID << 5, is_extended_id=False, data=data)
+                await self.try_send(msg)
+
+    async def try_send(self, msg):
+        try:
+            bus.send(msg)
+        except can.CanError:
+            LOGGER.error("Message NOT sent!  Please verify can0 is working first")
+
+    async def surface_errors(self):
+        for msg in bus:
+            if msg.arbitration_id == ((axisID << 5) | db.get_message_by_name('Heartbeat').frame_id):
+                errors = db.decode_message('Heartbeat', msg.data)['Axis_Error']
+                if errors != 0x0:
+                    await self.stop()
+                    LOGGER.error("axis:", ODriveError(errors).name)
+            
+                    msg = db.get_message_by_name('Clear_Errors')
+                    msg = can.Message(arbitration_id=msg.frame_id | axisID << 5, is_extended_id=False)
+                    await self.try_send(msg)
```

Comment:
just a little nit - remember to add a new line at the end of each file. a python linter like https://pypi.org/project/black/ will do this for you automatically, and lint things so that they're PEP8 compliant.